lastfmscrobbler/scrobble.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+
+set -euox pipefail
+
+tentativeMetadata=$(echo '{ "command": ["get_property", "metadata"] }' | /usr/bin/nc -U /tmp/mpv-socket)
+
+sleep 4
+
+USERNAME="" # CHANGE ME
+PASSWORD="" # CHANGE ME
+APIKEY="3176eb0bd0ff5d1c8f15d94e3b3c98a8"      # Change if the script stops working
+APISECRET="2918fac730f44f543d2568f9976ec276"   # Change if the script stops working
+
+lfmUrl="https://ws.audioscrobbler.com/2.0"
+
+metaDataRaw=$(echo '{ "command": ["get_property", "metadata"] }' | /usr/bin/nc -U /tmp/mpv-socket)
+durationRaw=$(echo '{ "command": ["get_property", "duration"] }' | /usr/bin/nc -U /tmp/mpv-socket)
+
+if [[ "$tentativeMetadata" != "$metaDataRaw" ]]; then
+  echo "Track skipped, not scrobbling" >&2
+  exit 1
+fi
+
+mediaTitle=$(echo "$metaDataRaw" | jq -r '.data.Title // .data.title // .data.TITLE // empty')
+mediaArtist=$(echo "$metaDataRaw" | jq -r '.data.Artist // .data.artist // .data.ARTIST // empty')
+mediaAlbum=$(echo "$metaDataRaw" | jq -r '.data.Album // .data.album // .data.ALBUM // empty')
+mediaAlbumArtist=$(echo "$metaDataRaw" | jq -r '.data.album_artist // .data.Album_artist // .data.ALBUM_ARTIST // .data.albumArtist // .data.albumartist // empty')
+
+mediaDuration=$(echo "$durationRaw" | jq -r '.data // empty')
+mediaTimestamp=$(date +%s)
+
+if [[ -z "$mediaTitle" ]]; then
+  echo "Couldn't find a title for this track in metadata, not scrobbling" >&2
+  exit 1
+fi
+if [[ -z "$mediaArtist" ]]; then
+  echo "Couldn't find an artist for this track in metadata, not scrobbling" >&2
+  exit 1
+fi
+if [[ -z "$mediaDuration" ]]; then
+  echo "Couldn't get track duration" >&2
+  exit 1
```

Comment:
no duration is kinda recoverable but like maybe it is better to exit bc something's rly broken if the socket isn't returning that